networking/external_dns_operator/nw-control-dns-records-public-hosted-zone-gcp.adoc

```diff
@@ -0,0 +1,153 @@
+= Controlling DNS records on an public hosted zone for GCP by using Red Hat External DNS Operator
+
+.Procedure
+
+. Export the kubeconfig file.
++
+[source,terminal]
+----
+$ export KUBECONFIG=~/<path to>/kubeconfig
+----
+
+. Check the user. The user must have access to the `kube-system` namespace if you don’t have the credentials as you can fetch the credentials from the `kube-system` namespace to use the cloud provider client.
++
+[source,terminal]
+----
+$ oc whoami
+----
++
+.Example output
+[source,terminal]
+----
+system:admin
+----
+
+. Copy the value of service_account.json in gcp-credentials secret in a file encoded-gcloud.json
++
+[source,terminal]
+----
+$ oc get secret gcp-credentials -n kube-system --template='{{$v := index .data "service_account.json"}}{{$v}}' | base64 -d - > decoded-gcloud.json
+----
+
+. Export Google credentials:
++
+[source,terminal]
+----
+$ export GOOGLE_CREDENTIALS=decoded-gcloud.json
+----
+
+. Activate your account:
++
+[source,terminal]
+----
+$ gcloud auth activate-service-account  <client_email as per decoded-gcloud.json> --key-file=decoded-gcloud.json
+----
+
+. Set your project
++
+[source,terminal]
+----
+$ gcloud config set project <project_id as per decoded-gcloud.json>
+----
+
+. Get the routes to check the domain
++
+[source,terminal]
+----
+$ oc get routes --all-namespaces | grep console
+----
++
+.Example output
+[source,terminal]
+----
+openshift-console          console             console-openshift-console.apps.qe.gcp.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.qe.gcp.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+----
+
+. Get your zone which was created by the installer
++
+[source,terminal]
+----
+$ gcloud dns managed-zones list | grep <dns name eg- As per the route the section after apps. i.e misalunk-azure.qe.azure.devcluster.openshift.com>
+----
+
+. Check the zone doesn't have DNS records other than `NS` and `SOA`
++
+[source,terminal]
+----
+$ gcloud dns record-sets list --zone=<the zone name you get from step 9>
+----
+
+. Create a ExternalDNS CR
++
+[source,yaml]
+----
+  apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-gcp <1>
+spec:
+  domains:
+    - filterType: Include <2>
+      matchType: Exact <3>
+      name: test-gcp1.qe.gcp.devcluster.openshift.com <4>
+  provider:
+    type: GCP <5>
+  source:
+    openshiftRouteOptions: <6>
+      routerName: default <7>
+    type: OpenShiftRoute <8>
+    fqdnTemplate:
+      - "{{.Name}}.apps.test-gcp1.qe.gcp.devcluster.openshift.com" <9>
+  #You will get this from step 9
+  zones:
+    - test-gcp1-q6m5v-private-zone <10>
+----
+
+<1> Name of external dns CR which will cause the operator to create a deployment and thereby the pods with this name
```

Comment:
Name of external dns CR which will cause the operator to create a deployment and thereby the external-dns pods in external-dns namespace with this name prefixed.